### ISSUE TITLE
ur_diver - waiting on server...

### ISSUE THREAD

**Issue by shaharlevy29** [ros-industrial/universal_robot#262]
I know this issue has been here already, but i've tried every solution suggested and i still cant get it to work.

Im following the tutorials for the Universal_robot package. I've established a connection 
recieved:

```
[INFO] [WallTime: 1474374444.432846] Robot connected
[2016-09-20 15:27:24.438572] Out: hello
```

but the `test_move.py` is stuck on waiting for server...
I tried using the one in `ur_modern_driver` with no luck.

My UR version is 3.218744 and im using Indigo.

The controller is fine, i can successfully send it commands using SocketTest.

Any suggestions?

**My bringup output:**

```
SUMMARY
========

PARAMETERS
 * /robot_description: <?xml version="1....
 * /robot_ip_address: 10.0.0.2
 * /robot_reverse_port: 50001
 * /rosdistro: indigo
 * /rosversion: 1.11.20
 * /tf2_buffer_server/buffer_size: 120.0
 * /ur_driver/max_payload: 10.0
 * /ur_driver/max_velocity: 10.0
 * /ur_driver/min_payload: 0.0
 * /ur_driver/prefix: 

NODES
  /
    robot_state_publisher (robot_state_publisher/state_publisher)
    tf2_buffer_server (tf2_ros/buffer_server)
    ur_driver (ur_driver/driver.py)

ROS_MASTER_URI=http://localhost:11311

core service [/rosout] found
process[robot_state_publisher-1]: started with pid [11201]
process[ur_driver-2]: started with pid [11202]
process[tf2_buffer_server-3]: started with pid [11203]
Setting prefix to 
[WARN] [WallTime: 1474374443.360245] No calibration offset for joint "shoulder_pan_joint"
[WARN] [WallTime: 1474374443.360513] No calibration offset for joint "shoulder_lift_joint"
[WARN] [WallTime: 1474374443.360718] No calibration offset for joint "elbow_joint"
[WARN] [WallTime: 1474374443.360914] No calibration offset for joint "wrist_1_joint"
[WARN] [WallTime: 1474374443.361104] No calibration offset for joint "wrist_2_joint"
[WARN] [WallTime: 1474374443.361292] No calibration offset for joint "wrist_3_joint"
[INFO] [WallTime: 1474374443.361537] No calibration offsets loaded from urdf
[INFO] [WallTime: 1474374443.362734] Max velocity accepted by ur_driver: 10.0 [rad/s]
[INFO] [WallTime: 1474374443.366584] Bounds for Payload: [0.0, 10.0]
Disconnected.  Reconnecting
[INFO] [WallTime: 1474374443.411940] Programming the robot
[INFO] [WallTime: 1474374443.414186] Programming the robot at 10.0.0.2
RobotStateRT has wrong length: 1060
Exception in thread URConnectionRT:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/arms_ros/catkin_ws/src/universal_robot/ur_driver/src/ur_driver/driver.py", line 372, in __run
    self.__on_packet(packet)
  File "/home/arms_ros/catkin_ws/src/universal_robot/ur_driver/src/ur_driver/driver.py", line 338, in __on_packet
    for i, q in enumerate(stateRT.q_actual):
AttributeError: 'RobotStateRT' object has no attribute 'q_actual'

Likely a syntax error:
4���������  URControlI8Oct 07 2015, 11:12:33
Exception in thread URConnection:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/arms_ros/catkin_ws/src/universal_robot/ur_driver/src/ur_driver/driver.py", line 274, in __run
    self.__on_packet(packet)
  File "/home/arms_ros/catkin_ws/src/universal_robot/ur_driver/src/ur_driver/driver.py", line 186, in __on_packet
    state = RobotState.unpack(buf)
  File "/home/arms_ros/catkin_ws/src/universal_robot/ur_driver/src/ur_driver/deserialize.py", line 324, in unpack
    raise Exception("Fatal error when unpacking RobotState packet")
Exception: Fatal error when unpacking RobotState packet

[INFO] [WallTime: 1474374444.417094] Programming the robot at 10.0.0.2
Handling a request
[INFO] [WallTime: 1474374444.432846] Robot connected
[2016-09-20 15:27:24.438572] Out: hello
Connection closed (command): EOF on recv
Handling a request
[2016-09-20 15:27:24.910425] Out: hello
```

**Comment by miguelprada** [comment]
Which driver are you using? The [`ur_driver`](https://github.com/ros-industrial/universal_robot) or the [`ur_modern_driver`](https://github.com/ThomasTimm/ur_modern_driver)? I'm guessing the latter, since the `ur_driver` is, AFAIK, not compatible with UR v3.X.

Could you provide a more detailed description of the exact steps you're following?

**Comment by gavanderhoorn** [comment]
There is currently no point in trying to use `ur_driver` with anything over 1.8.x, so let's ignore that.

If you can test with the `ur_modern_driver` and run into the same problems, then please report this to @ThomasTimm over at `ur_modern_driver`.

**Comment by gavanderhoorn** [comment]
@miguelprada: from the exceptions it would seem this is actually `ur_driver`.

**Comment by miguelprada** [comment]
@gavanderhoorn There's more detail in the original post now. Guess @shaharlevy29 edited after I replied.

Close this?

**Comment by gavanderhoorn** [comment]
Let's give @shaharlevy29 an opportunity to respond first.

But it would seem like this is a `WONTFIX` for now, at least for `ur_driver`.

**Comment by gavanderhoorn** [comment]
@shaharlevy29: :bell: ?

**Comment by gavanderhoorn** [comment]
Closing because of inactivity.

@shaharlevy29: if you have any updates, or things are still not working for you, please re-open.

**Comment by philipsgithub** [comment]
@shaharlevy29 @gavanderhoorn I had the same errors (no attribute q_actual and fatal error when unpacking RobotState).

If somebody still has similar issues:

1. Make sure the robot is actively sending its status via tcp when a client is connected.
--> Test this with TestSocket3.0 or similar. Just connect to ur robot ip and port 30002. There should be a steady stream of incomming packets.

2. If your robot is sending its status, there is hope:) No lets come to the errors. The errors mentioned above (q_actual and unpacking problems) are not the real problem. The real problem is: "RobotStateRT has wrong length: 1060". Do a text search with this error in your universal_robot package and you will find it in deserializeRT.py. The length 1060 is unknown and belongs to the new protocol version 3.4 (see the xlsx-file at https://www.universal-robots.com/how-tos-and-faqs/how-to/ur-how-tos/remote-control-via-tcpip-16496/ for more information).

3. Quickest fix (on your own responsibility): Just add to the if-statements in the unpack method in deserializeRT.py:   
elif plen == 1060: 
            return RobotStateRT_V30.unpack(buf)

Notes: You don't get the OutputStates and the ProgramState (new in v3.4) when you do it this way.

**Comment by htugal** [comment]
I have the same problem while testing for UR3, yet can't figure out how to solve the problem. Any suggestion ?

**Comment by gavanderhoorn** [comment]
Don't use `ur_driver` with UR3s. Please see the `readme` in the repository root and the *Compatibility* section on the [wiki/universal_robot](http://wiki.ros.org/universal_robot) page.

**Comment by htugal** [comment]
Actually I am using with ur_modern_driver

**Comment by gavanderhoorn** [comment]
Then you should not post your issue on this tracker.

Please post it on the issue tracker of `ur_modern_driver`.

**Comment by ProfJust** [comment]
Got the same Problems described above. `ur_modern_driver `is deprecated by now.
Since I use  `UniversalRobots/Universal_Robots_ROS_Driver` including the `fmauch/universal_robot` - repository which is replacing  `ros-industrial/universal_robot ` this problem ist solved.

**Comment by gavanderhoorn** [comment]
> Since I use `UniversalRobots/Universal_Robots_ROS_Driver` including the `fmauch/universal_robot` - repository which is replacing `ros-industrial/universal_robot ` this problem ist solved.

I'm glad you got things to work, but some clarification: `fmauch/universal_robot` will be merged back into `ros-industrial/universal_robot` so this is only a temporary situation.
